imageproxy.go

```diff
@@ -203,6 +211,20 @@ func validHost(hosts []string, u *url.URL) bool {
 	return false
 }
 
+// blacklistedHost returns true if the host is in our blacklist
+func blacklistedHost(hosts []string, u *url.URL) bool {
```

Comment:
You are most certainly correct about that. Late night hack (from a non-developer such as me) that turned into something my colleagues insisted could be useful for others :) 

I will fix all 3 issues you pointed out tomorrow morning and do another pull request (or update this one if possible...)

I also managed to reduce the resulting Docker image from 703.5 MB to 6.7 MB. I can add a script that builds that image maybe? Its a two stage docker build process so a single dockerfile is not enough.